packages/grafana-ui/src/components/FilterPill/FilterPill.tsx

```diff
@@ -13,13 +13,18 @@ export interface FilterPillProps {
   label: string;
   onClick: React.MouseEventHandler<HTMLElement>;
   icon?: IconName;
+  customClass?: string;
```

Comment:
>  set a custom class for further customization from the component using it

This is explicitly an anti-goal for @grafana/ui. "Further customization" means more inconsistency, and more unreliability as implementation details of the components are iterated on.

For your use case, I think it's fine to add `white-space: nowrap` directly to this component. I don't think FilterPill should be used for long content that we want to wrap, and it appears the only use of it is in Transformations. Doesn't look like its used in plugins we know about.

Reply:
agree, and customClass is also the wrong way to solve it if we where to allow customization (it should be done with className and forwardRef)

Reply:
removed the customClass